scheduler/src/background/sub_job_scaling.rs

```diff
@@ -0,0 +1,213 @@
+        _ => Err(SubJobHandlerError::FailedJob(
+            "Invalid and unexpected status".to_string(),
+        )),
+    };
+
+    match result {
+        Ok(_) => {}
+        Err(SubJobHandlerError::Skip(e)) => {
+            info!("SubJobScalingError::Skip: {}", e);
+        }
+        Err(SubJobHandlerError::FailedJob(e)) => {
+            error!("ubJobScalingError::FailedJob: {}", e);
+
+            let _ = repo
+                .sub_job
+                .update_sub_job_status_with_error(&sub_job.id, SubJobStatus::Failed, e)
+                .await;
+        }
+    }
+
+    Ok(())
+}
+
+#[tracing::instrument(skip(repo, service_scaler_registry, sub_job), fields(sub_job_id = %sub_job.id))]
+async fn process_scaling_created(
+    repo: Arc<Repositories>,
+    service_scaler_registry: Arc<ServiceScalerRegistry>,
+    sub_job: &SubJob,
+) -> Result<(), SubJobHandlerError> {
+    info!("Processing scaling created type sub job");
+
+    let topic = get_topic(sub_job)?;
+    let services = get_services(repo.clone(), &topic).await?;
+    let workers_online = get_workers_online(repo.clone(), &topic).await?;
+    let service_count = services.len();
+
+    let scale_each_by = MAX_JOB_WORKERS.div_ceil(service_count);
+
+    // Set expected worker count into the job details
+    let workers_count = (scale_each_by * service_count) as i64;
+    repo.job
+        .update_job_workers_count(&sub_job.job_id, workers_count)
+        .await
+        .map_err(|e| SubJobHandlerError::Skip(e.to_string()))?;
+
+    // Extend the descale_at deadline for all services
+    let descale_at = Utc::now() + Duration::from_secs(SERVICE_DEADLINE_SEC);
+    repo.service
+        .set_descale_deadlines(&services.iter().map(|s| s.id).collect(), descale_at)
+        .await
+        .map_err(|e| SubJobHandlerError::Skip(e.to_string()))?;
+
+    debug!(
+        "workers_online: {} services: {:?}, scale_each_by: {}, service_count: {}, workers_count: {}, descale_at: {}",
+        workers_online.len(),
+        services.iter().map(|s| &s.id),
+        scale_each_by,
+        service_count,
+        workers_count,
+        descale_at
+    );
+
+    // Do not scale if workers are already online
+    // Set the sub job status to canceled
+    if workers_online.len() >= service_count {
+        repo.sub_job
+            .update_sub_job_status(&sub_job.id, SubJobStatus::Canceled)
+            .await
+            .map_err(|e| SubJobHandlerError::Skip(e.to_string()))?;
+
+        return Ok(());
+    }
+
+    //  TODO: we are lacking a diff...
+    //  service picker doesnt care about even distribution of services by the topic
+    //  if we run poland that has europe topic then europe topic will skip, and will not spawn additional instances or the rest of services with europe topic
+    //  also we do not count diff of online and max versus missing workers, that might spawn more workers than we need
```

Comment:
That's OK. I don't expect users to mix "granularities". They will either run for topics europe + us-east or for topics poland + frankfurt.